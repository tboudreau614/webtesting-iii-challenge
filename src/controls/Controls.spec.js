import React from 'react';
import * as rtl from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Controls from './Controls'

test('Controls render as expected', () =>{
    const panel = rtl.render(<Controls/>)
    expect (panel.asFragment()).toMatchSnapshot()
})

test('Buttons render as expected', () => {
    const panel = rtl.render(<Controls/>)
    const lock = panel.getByText(/lock gate/i)
    const open = panel.getByText(/close gate/i)
    expect(lock).toBeDefined();
    expect(open).toBeDefined();
})

test('If lock is currently in use, user should not be able to make the button fire again', () => {
    const mock = jest.fn()
    const panel = rtl.render(<Controls toggleLocked={mock}/>)
    const button = panel.getByText(/lock gate/i);
    rtl.act(() => {rtl.fireEvent.click(button) })
	expect(mock).toHaveBeenCalledTimes(0);
})