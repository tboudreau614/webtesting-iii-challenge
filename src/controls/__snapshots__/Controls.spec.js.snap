// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls (buttons) render without crashing 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;

exports[`Controls render as expected 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;

exports[`Controls render without crashing 1`] = `
<DocumentFragment>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
